File: backend/src/main/java/com/careerpath/application/dto/ProfileDto.java
package com.careerpath.application.dto;

import java.util.List;

public record ProfileDto(
        String fullName,
        String headline,
        String about,
        String location,
        List<ProfileSkillDto> skills,
        List<ProfileExperienceDto> experiences,
        boolean aiOptIn
) { }

File: frontend/src/app/core/models/profile.dto.ts
import {ProfileSkill} from './profile-skill';
import {ProfileExperience} from './profile-experience';

export interface ProfileDto {
  fullName: string;
  headline: string;
  about: string;
  location: string;
  skills: ProfileSkill[];
  experiences: ProfileExperience[];
  aiOptIn: boolean;
}
